### PR TITLE
CSS hacks for Web Developers

### LINKED ISSUE #2819: CSS Hacks Every Web Developer Should Know
### Proposed title of article
CSS Hacks Every Web Developer Should Know

### Introduction paragraph (2-3 paragraphs):
CSS is still among the top ten coding languages, according to Redmonk, despite not being as popular as JavaScript.
CSS is popular among website developers because it is quite robust, relatively straightforward to learn, and ubiquitous across different browsers.

There are various CSS shortcuts or hacks, as there are with any coding language, that help you to write cleaner code, improve design components, and save time. 

It's also vital to note that CSS does not require you to be a senior web developer.
CSS is utilized by 96 percent of all websites, according to W3Techs, and the ability to use CSS to improve the layout and design of a website is essential to the functionality of major open source CMSs like WordPress. 

If you’re new to CSS, this article will teach you the fundamental aspects, and CSS hacks that will be beneficial in your work. 

### Key takeaways: 
1. How to position content in the center with CSS
2. How to fit images to the page in CSS
3. How to consolidate styling in CSS
4. Visited link styling in CSS
5. How to disable text wrapping and add ellipsis in CSS.

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by mosesmoses12**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2819

### CODE REVIEW

**Comment by geoffrey45** [comment]
@mosesmoses12 Welcome to the EngEd program 🎉. Thank you for submitting your article for review.

**Comment by mosesmoses12** [comment]
Thanks for the ping @geoffrey45, looking forward to cooperate.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/css-hacks-for-web-developers/index.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  width: 500px;
+  height: 500px;
+  object-fit: scale-down;
+ }
+```
+In the above snippet, the `object-fit attribute specifies how an `image should be scaled to fit within its container.
+
+4. ### CSS style for visited links
+These styles are very important, yet some developers decide to avoid them. This sometimes causes trouble to the end-users as far as the usability of the web app is concerned.
+
```
Comment (geoffrey45) [inline content/articles/css-hacks-for-web-developers/index.md:77]: This line is ambiguous. Please provide examples of usability troubles that can be caused

File: content/articles/css-hacks-for-web-developers/index.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+```html
+<html>
+  <body>
+    <h4>Why do developer Group elements together when styling?</h4>
+    <p>Developers Group elements together to save on time </p>
+    <h5>What are advantages Of Consolidating during styling?</h5>
+    <style>
+      /* suppose you want to add a background color, text color, font and center the 3 elements in the middle of the screec */
+      h4,p,h5{
+        background-color: black;
```
Comment (geoffrey45) [inline content/articles/css-hacks-for-web-developers/index.md:135]: In cases where different elements within different classes are the target, this method is not convenient. Please consider using classes and ids for this.

File: content/articles/css-hacks-for-web-developers/index.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+}
+/* color green represent visited link */
+a:visited {
+  color: green;
+}
+/*color blue represent selected link */
+a:active {
+  color: blue
+```
+5. ### How to style Initial Letter using CSS
```
Comment (geoffrey45) [inline content/articles/css-hacks-for-web-developers/index.md:96]: `How to style the first letter in a paragraph` sounds better

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@mosesmoses12 Thank you for submitting your article for review. Please check the comments below.

**Comment by mosesmoses12** [comment]
@geoffrey45, Am done with the requested changes. Please have a look at my last commit.

**Review by geoffrey45 — APPROVED** [review]

**Comment by geoffrey45** [comment]
@mosesmoses12 Please check the changes I made in [this commit](https://github.com/section-engineering-education/engineering-education/pull/3105/commits/ed7c5f6a8bc84f461dbc8a8bec9e866f93cc943c).

Thank you for contributing to the EngEd community.

**Comment by hectorkambow** [comment]
Overall nice work gents 🚀

**Comment by hectorkambow** [comment]
Be sure to go over any suggestions / changes made by your peer reviewer to improve future articles.

**Pull request merged**